numad: 0.5 -> 0.5-unstable-2023-09-06

VincentVanlaer opened the pull request:
## Description of changes

numad is broken in nixpkgs  due to a buffer overrun in the logging routine. This only became a visible problem since FORTIFY_SOURCE got enabled by default. The bug has been fixed upstream. As it doesn't appear that there are proper tagged releases, I have just taken the version from the man page.

The diff between the current version in nixpkgs and the latest upstream are too much to go through manually. For what it's worth, both debian and fedora package this version (or close to it) in their respective stable releases.

Given that the package is currently broken, I think it make sense to also backport the update.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

VincentVanlaer committed:
numad: 0.5-334278ff -> 1.0.0-3399d893

pbsds commented on pkgs/os-specific/linux/numad/default.nix:
```diff
@@ -2,19 +2,14 @@
 stdenv.mkDerivation rec {
   pname = "numad";
-  version = "0.5";
+  version = "1.0.0";
```
```suggestion
  version = "1.0.0-unstable-2023-09-06";
```

pbsds reviewed (approved):
Result of `nixpkgs-review pr 315168` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages marked as broken and skipped:</summary>
  <ul>
    <li>libvmi</li>
    <li>nixops_unstablePlugins.nixops-libvirtd</li>
    <li>nixops_unstablePlugins.nixops-libvirtd.dist</li>
    <li>nixops_unstablePlugins.nixops-virtd</li>
    <li>nixops_unstablePlugins.nixops-virtd.dist</li>
    <li>nixops_unstable_full</li>
    <li>nixops_unstable_full.dist</li>
  </ul>
</details>
<details>
  <summary>40 packages built:</summary>
  <ul>
    <li>appvm</li>
    <li>collectd</li>
    <li>diffoscope</li>
    <li>diffoscope.dist</li>
    <li>diffoscope.man</li>
    <li>docker-machine-kvm2</li>
    <li>gnome.gnome-boxes</li>
    <li>guestfs-tools</li>
    <li>htcondor</li>
    <li>libguestfs</li>
    <li>librenms</li>
    <li>libvirt</li>
    <li>libvirt-glib</li>
    <li>libvirt-glib.dev</li>
    <li>libvirt-glib.devdoc</li>
    <li>mgmt</li>
    <li>minikube</li>
    <li>multipass</li>
    <li>numad</li>
    <li>perl536Packages.SysVirt</li>
    <li>perl536Packages.SysVirt.devdoc</li>
    <li>perl538Packages.SysVirt</li>
    <li>perl538Packages.SysVirt.devdoc</li>
    <li>python311Packages.guestfs</li>
    <li>python311Packages.guestfs.dist</li>
    <li>python311Packages.libvirt</li>
    <li>python311Packages.libvirt.dist</li>
    <li>python312Packages.guestfs</li>
    <li>python312Packages.guestfs.dist</li>
    <li>python312Packages.libvirt</li>
    <li>python312Packages.libvirt.dist</li>
    <li>rubyPackages.ruby-libvirt (rubyPackages_3_1.ruby-libvirt)</li>
    <li>rubyPackages_3_2.ruby-libvirt</li>
    <li>rubyPackages_3_3.ruby-libvirt</li>
    <li>vagrant</li>
    <li>virt-manager</li>
    <li>virt-manager-qt</li>
    <li>virt-manager.dist</li>
    <li>virt-top</li>
    <li>virt-viewer</li>
  </ul>
</details>

---


This catapults us forward 12 years.

They did indeed switch to 1.0.0 [11 years ago](https://pagure.io/numad/c/a1f2291dd92e5843dd93956312f280934ed24ad9), which no-one else seems to have picked up on https://repology.org/project/numad/versions

There is also the VERSION_STRING [here](https://pagure.io/numad/blob/master/f/numad.c#_60) stuck at `20150602` which debian reports.

pbsds commented:
Please also add `passthru.updateScript = unstableGitUpdater { };`

VincentVanlaer committed:
numad: enable unstableGitUpdater

The upstream repository doesn't have tags, so the base version is "0".

VincentVanlaer commented:
I had to change the base version to "0", since there are no upstream tags. While not ideal, it's probably preferable over not having notifications of new version.

pbsds commented on pkgs/os-specific/linux/numad/default.nix:
```diff
@@ -1,26 +1,23 @@
 
   makeFlags = [ "prefix=$(out)" ];
 
+  passthru.updateScript = unstableGitUpdater { };
```
The version `0` will not match the [other distros](https://repology.org/project/numad/versions), what about

```suggestion
  passthru.updateScript = unstableGitUpdater {
    tagConverter = ''( read tag; test "$tag" = "0" && tag=0.5; echo "$tag" )'';
  };
```

VincentVanlaer replied:
Thanks for  the suggestion! I had to put it behind writeShellScript (bash didn't seem to like the brackets), but otherwise it worked well

VincentVanlaer committed:
numad: set base version to 0.5 using tag converter

Since the upstream repository doesn't have any tags set, the
unstable git updater will use 0 as a base version.

pbsds commented on pkgs/os-specific/linux/numad/default.nix:
```diff
@@ -1,26 +1,25 @@
   makeFlags = [ "prefix=$(out)" ];
 
+  passthru.updateScript = unstableGitUpdater {
+    tagConverter = writeShellScript "tagConverter" ''( read tag; test "$tag" = "0" && tag=0.5; echo "$tag" )'';
```
Well into nit-territory now, but we can save ourselves a sub-shell
```suggestion
    tagConverter = writeShellScript "tagConverter" ''
      read tag; test "$tag" = "0" && tag=0.5; echo "$tag"
    '';
```

VincentVanlaer replied:
Should be fixed now
